File: src/import/ScotlandsMountains.Import.ConsoleApp/Models/Mountain.cs
namespace ScotlandsMountains.Import.ConsoleApp.Models;

internal class Mountain : Entity
{
    public decimal Latitude { get; set; }
    public decimal Longitude { get; set; }
    public string GridRef { get; set; } = null!;
    public decimal Height { get; set; }
        
    public string? Feature { get; set; }
    public string? Observations { get; set; }

    public decimal Drop { get; set; }
    public string Col { get; set; } = null!;
    public decimal ColHeight { get; set; }

    public MountainSummary Parent { get; set; } = null!;
    public IList<string> Aliases { get; set; } = null!;

    public IList<Entity> Classifications { get; set; } = new List<Entity>();
    public IList<Entity> Countries { get; set; } = new List<Entity>();

    public int DobihId { get; set; }
}

File: src/scotlands_mountains_app/lib/models/mountain.dart
class Mountain {
  final int id;
  final String name;
  final double latitude;
  final double longitude;
  final String gridRef;
  final double height;
  final String? feature;
  final String? observations;
  final double drop;
  final String col;
  final double colHeight;
  final int dobihId;

  Mountain._(
      this.id,
      this.name,
      this.latitude,
      this.longitude,
      this.gridRef,
      this.height,
      this.feature,
      this.observations,
      this.drop,
      this.col,
      this.colHeight,
      this.dobihId);

  Mountain(Map<String, dynamic> map)
      : this._(
            map['id'],
            map['name'],
            map['latitude'],
            map['longitude'],
            map['gridRef'],
            map['height'],
            map['feature'],
            map['observations'],
            map['drop'],
            map['col'],
            map['colHeight'],
            map['dobihId']);
}
